imports/plugins/core/graphql/server/resolvers/account/Mutation/addAccountAddressBookEntry.js

```diff
@@ -1,23 +1,24 @@
-import { decodeAccountOpaqueId, encodeAccountOpaqueId } from "@reactioncommerce/reaction-graphql-xforms/account";
+import { decodeAccountOpaqueId } from "@reactioncommerce/reaction-graphql-xforms/account";
+import { xformAddressResponse } from "@reactioncommerce/reaction-graphql-xforms/address";
 
 /**
  * @name addAccountAddressBookEntry
  * @method
  * @summary resolver for the addAccountAddressBookEntry GraphQL mutation
  * @param {Object} _ - unused
- * @param {Object} args - an object of all arguments that were sent by the client
+ * @param {Object} args.input - an object of all mutation arguments that were sent by the client
+ * @param {String} [args.input.accoundId] - The account ID, which defaults to the viewer account
+ * @param {AddressInput} args.input.address - The address to add
+ * @param {String} [args.input.clientMutationId] - An optional string identifying the mutation call
  * @param {Object} context - an object containing the per-request state
  * @return {Object} AddAccountAddressBookEntryPayload
  */
 export default function addAccountAddressBookEntry(_, { input }, context) {
   const { accountId, address, clientMutationId = null } = input;
-  const dbAccountId = decodeAccountOpaqueId(accountId);
+  const dbAccountId = accountId && decodeAccountOpaqueId(accountId);
   const updatedAddress = context.methods["accounts/addressBookAdd"](context, [address, dbAccountId]);
   return {
-    address: {
-      ...updatedAddress,
-      _id: encodeAccountOpaqueId(updatedAddress._id)
-    },
+    address: xformAddressResponse(updatedAddress),
```

Comment:
See my comment about the method return value. If you change that to return false, then you should check here. I'm thinking `if (!updatedAddress)` then you should probably throw an error so that it's clear to the client that the add was unsuccessful. It'll have to be a rather vague error message because we won't know why it failed, but it's also unlikely to happen.

Reply:
I threw the error in "accounts/addressBookAdd", instead of returning `false`, so if the update failed for some rare reason, this we'll never get to this point and the error will show in the response.